src/geometry.rs

```diff
@@ -1671,4 +1745,93 @@ mod tests {
+            Point3::new(-0.866025, -0.5, 0.0),
+            Point3::new(0.866025, -0.5, 0.0),
+        ];
+        let geometry = Geometry::from_triangle_faces_with_vertices_and_computed_normals(
+            faces,
+            vertices,
+            NormalStrategy::Sharp,
+        );
+
+        let normal_correct = Vector3::new(0.0, 0.0, 1.0);
+
+        let Face::Triangle(t_f) = geometry.faces()[0];
+        let normal_calculated = compute_triangle_normal(
+            &geometry.vertices()[cast_usize(t_f.vertices.0)],
+            &geometry.vertices()[cast_usize(t_f.vertices.1)],
+            &geometry.vertices()[cast_usize(t_f.vertices.2)],
+        );
+
+        assert_eq!(normal_correct, normal_calculated);
+    }
+
+    #[test]
+    fn test_geometry_compute_triangle_normal_returns_x_vector_for_vertical_triangle() {
+        let faces = vec![(0, 1, 2)];
+        let vertices = vec![
+            Point3::new(0.0, 1.0, 0.0),
+            Point3::new(0.0, -0.5, 0.866025),
+            Point3::new(0.0, -0.5, -0.866025),
+        ];
+        let geometry = Geometry::from_triangle_faces_with_vertices_and_computed_normals(
+            faces,
+            vertices,
+            NormalStrategy::Sharp,
+        );
+
+        let normal_correct = Vector3::new(1.0, 0.0, 0.0);
+
+        let Face::Triangle(t_f) = geometry.faces()[0];
+        let normal_calculated = compute_triangle_normal(
+            &geometry.vertices()[cast_usize(t_f.vertices.0)],
+            &geometry.vertices()[cast_usize(t_f.vertices.1)],
+            &geometry.vertices()[cast_usize(t_f.vertices.2)],
+        );
+
+        assert_eq!(normal_correct, normal_calculated);
+    }
+
+    #[test]
+    fn test_geometry_compute_triangle_normal_returns_vector_for_arbitrary_triangle() {
+        let faces = vec![(0, 1, 2)];
+        let vertices = vec![
+            Point3::new(0.268023, 0.8302, 0.392469),
+            Point3::new(-0.870844, -0.462665, 0.215034),
+            Point3::new(0.334798, -0.197734, -0.999972),
+        ];
+        let geometry = Geometry::from_triangle_faces_with_vertices_and_computed_normals(
+            faces,
+            vertices,
+            NormalStrategy::Sharp,
+        );
+
+        let normal_correct = Vector3::new(0.62270945, -0.614937, 0.48382375);
+
+        let Face::Triangle(t_f) = geometry.faces()[0];
+        let normal_calculated = compute_triangle_normal(
+            &geometry.vertices()[cast_usize(t_f.vertices.0)],
+            &geometry.vertices()[cast_usize(t_f.vertices.1)],
+            &geometry.vertices()[cast_usize(t_f.vertices.2)],
+        );
+
+        assert_eq!(normal_correct, normal_calculated);
+    }
+
+    #[test]
+    fn test_geometry_plane_new_calculate_perpendicular_y() {
+        let test_plane = Plane::new(
+            &Point3::new(0.0, 0.0, 0.0),
+            &Vector3::new(1.0, 0.0, 0.0),
+            &Vector3::new(1.0, 1.0, 0.0),
+        );
+
+        assert_eq!(test_plane.y_vector, Vector3::new(0.0, 1.0, 0.0));
```

Comment:
We have no new agreement beyond what you already know. Skimming through the changes in this PR, this would be a good candidate for property testing, which we de-prioritized. The next best thing is to test all edge-cases and one average case (I think that's the same as what you suggest)

Reply:
There are too many edge cases here. I decided to test some and hope for the rest.

Reply:
That's fine, but let's test those that you explicitly assert in the plane constructor - it means you have already thought about those 😉